File: src/main/java/no/nav/syfo/rest/domain/RSKontaktinfo.java
package no.nav.syfo.rest.domain;


import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

@Data
@Accessors(fluent = true, chain = true)
@EqualsAndHashCode
public class RSKontaktinfo {

    public String fnr;
    public String epost;
    public String tlf;
    public Boolean skalHaVarsel;
    public FeilAarsak feilAarsak;


    public enum FeilAarsak {
        RESERVERT,
        UTGAATT,
        KONTAKTINFO_IKKE_FUNNET,
        SIKKERHETSBEGRENSNING,
        PERSON_IKKE_FUNNET
    }
}

File: src/main/kotlin/no/nav/syfo/kontaktinfo/controller/RSKontaktinfo.kt
package no.nav.syfo.kontaktinfo.controller

data class RSKontaktinfo(
    val fnr: String,
    val epost: String? = null,
    val tlf: String? = null,
    val skalHaVarsel: Boolean,
    val feilAarsak: FeilAarsak? = null
)

enum class FeilAarsak {
    RESERVERT, UTGAATT, KONTAKTINFO_IKKE_FUNNET, SIKKERHETSBEGRENSNING, PERSON_IKKE_FUNNET
}
